server/src/main/java/org/opensearch/snapshots/SnapshotUtils.java

```diff
@@ -135,4 +143,36 @@ public static List<String> filterIndices(List<String> availableIndices, String[]
         }
         return Collections.unmodifiableList(new ArrayList<>(result));
     }
+
+    /**
+     * Validates if there are any remote snapshots backing an index
+     * @param currentState current cluster state
+     * @param snapshotIds list of snapshot Ids to be verified
+     * @param repoName repo name for which the verification is being done
+     */
+    public static void validateSnapshotsBackingAnyIndex(ClusterState currentState, List<SnapshotId> snapshotIds, String repoName) {
+        final Map<String, SnapshotId> uuidToSnapshotId = new HashMap<>();
+        final Set<String> snapshotsToBeNotDeleted = new HashSet<>();
+        ImmutableOpenMap<String, IndexMetadata> indicesMap = currentState.getMetadata().getIndices();
```

Comment:
Since all you need is the index metadata, I wouldn't pass in the whole cluster state object. Either pass in `currentState.getMetadata().indices()` or even `currentState.getMetadata().indices().values()`. It should make the unit tests simpler too as you don't need to construct a whole cluster state object.

Reply:
Now only passing indices metadata